geometry/scalar.h

```diff
@@ -5,13 +5,24 @@
 #pragma once
 
 #include <cfloat>
+#include <limits>
+#include <valarray>
 
+#include "flutter/fml/logging.h"
 #include "impeller/geometry/constants.h"
 
 namespace impeller {
 
 using Scalar = float;
 
+constexpr bool ScalarNearlyEqual(
+    Scalar x,
+    Scalar y,
+    Scalar tolerance = std::numeric_limits<Scalar>::epsilon() * (1 << 16)) {
```

Comment:
Isn't this default tolerance too small? Perhaps just 1e-3?

Reply:
Yeah, this seems much more reasonable. Done.

Reply:
Though, I should say I didn't know numeric_limits<T>::epsilon was a thing. I would have defaulted to FLT_EPSILON. Learnt something new.